src/thorlabs_cube/driver/tcube/tsc.py

```diff
@@ -0,0 +1,346 @@
+        """
+        payload = st.pack(
+            "<HHllHH", self._CHANNEL, mode, position1, position2, timeout1, timeout2
+        )
+
+        await self.send(Message(MGMSG.MOT_SET_BUTTONPARAMS, data=payload))
+
+    async def get_button_parameters(self) -> tuple[int, int, int, int, int]:
+        """Get the current button parameters.
+
+        :return: A tuple containing (mode, position1, position2, timeout1, timeout2)
+        """
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_BUTTONPARAMS, [MGMSG.MOT_GET_BUTTONPARAMS], self._CHANNEL
+        )
+
+        return st.unpack("<HllHH", get_msg.data[2:])
+
+    async def set_eeprom_parameters(self, msg_id: int) -> None:
+        """Save the current parameters for the specified message in the EEPROM.
+
+        This function sends a request to the device to save the parameter settings
+        of the specified message to the EEPROM memory, ensuring the settings are
+        retained even after the device is powered off.
+
+        :param msg_id: The message ID of the message containing the parameters
+                       that need to be saved in the EEPROM.
+        """
+        payload = st.pack("<HH", self._CHANNEL, msg_id)
+
+        await self.send(Message(MGMSG.MOT_SET_EEPROMPARAMS, data=payload))
+
+    async def get_status_update(self) -> tuple[int, int, int, int]:
+        """Returns a status update on the specified motor channel
+
+        This function sends a request to the device to return status updates of
+        position, encoder count, status bits on channel one as well as, for future use,
+        returns channel identity two and its associated data which is garbage right now
+
+        :return: A tuple containing (position, encoder_count, status_bits, chan_identity_two)
+        """
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_STATUSUPDATE, [MGMSG.MOT_GET_STATUSUPDATE], self._CHANNEL
+        )
+        (
+            position,
+            encoder_count,
+            status_bits,
+            chan_identity_two,
+            _,
+            _,
+            _,
+        ) = st.unpack("<HIIIHIII", get_msg.data[2:])
+
+        return position, encoder_count, status_bits, chan_identity_two
+
+    async def set_sol_operating_mode(self, operating_mode: int) -> None:
+        """Set the solenoid operating mode for the single channel.
+
+        :param operating_mode: The operating mode to set (e.g., 1, 2, etc.).
+        """
+        await self.send(Message(MGMSG.MOT_SET_SOL_OPERATINGMODE, param2=operating_mode))
+
+    async def get_sol_operating_mode(self) -> int:
+        """Get the current solenoid operating mode for the single channel.
+
+        :return: Operating mode of solenoid represented by an integer
+        """
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_SOL_OPERATINGMODE,
+            [MGMSG.MOT_GET_SOL_OPERATINGMODE],
+            param1=self._CHANNEL,
```

Comment:
```suggestion
            param1=Tsc._CHANNEL,
```
Use class constants.